packages/flutter_test/lib/src/window.dart

```diff
@@ -646,20 +671,21 @@ class TestFlutterView implements FlutterView {
   /// See also:
   ///
   ///   * [FlutterView.devicePixelRatio] for the standard implementation
+  ///   * [TestDisplay.devicePixelRatio] which will stay in sync with this value
   ///   * [resetDevicePixelRatio] to reset this value specifically
   ///   * [reset] to reset all test values for this view
   @override
-  double get devicePixelRatio => _devicePixelRatio ?? _view.devicePixelRatio;
-  double? _devicePixelRatio;
+  double get devicePixelRatio => _display._devicePixelRatio ?? _view.devicePixelRatio;
```

Comment:
The DPR of the view and its display must be the same. Consider adding an assertion that they are the same.

Reply:
I don't see how an assertion adds any value here? The default values come straight from the engine, and these test classes are set up such that you _can't_ change just a View or just a Display. Updating one will update the other. What kind of assertion are you looking for?

Reply:
An assertion that they're the same :)

Reply:
Alright, well that was not a very helpful clarification.

I considered adding an assertion that the values from the engine are equal, but that would create false negatives for tests using either `TestFlutterView.devicePixelRatio` or `TestDisplay.devicePixelRatio` should there be a bug in the engine. Further considering this is specifically written to prevent a user from setting one of those values in tests without setting the other, and considering that there are tests for both `TestFlutterView` and `TestDisplay` to validate that one cannot be set without setting the other to avoid issues with possible future changes, I'd rather not add an assertion.

If you have a specific request of what you'd like asserted, and if that request would not lead to false negatives for tests if there's a bug in the engine, then I'd be happy to revisit this at a later date.

Reply:
I was misreading what `_view` was. I see how this works now.